### PR TITLE
Integrated GraphQL Zero API

### PR DESCRIPTION
**Pull request by glarence**
GraphQL Zero integrated with new Combobox search bar with search suggestion dropdown, also have basic setup to reintegrate UserProfile display when selecting a user from search

### CODE REVIEW

**Comment by glarence** [comment]
GraphQL Zero integrated with new Combobox search bar with search suggestion dropdown, also have basic setup to reintegrate UserProfile display when selecting a user from search

File: basicapp/src/App.js [hunk 1]
```diff
@@ -2,40 +2,65 @@ import logo from './logo.svg';
...
+function Users({ query, setQuery }) {
+  const { loading, error, data } = useQuery(GET_USERS);
 
-const filterUsernames = (users, query) => {
-  if(!query){
-    console.log("invalid query, won't filter");
-    return users;
+  if (loading){
+    console.log("Loading API call!");
+    //return <p>Loading...</p>;
```
Comment (zhangxian97) [inline basicapp/src/App.js:29]: remove unused comments

File: basicapp/src/App.js [hunk 2]
```diff
@@ -2,40 +2,65 @@ import logo from './logo.svg';
...
+    return 'Loading...';
+  }
+  if (error){
+    console.log("Error with API call?");
+    //return <p>Error :(</p>;
```
Comment (zhangxian97) [inline basicapp/src/App.js:34]: same.

File: basicapp/src/App.js [hunk 3]
```diff
@@ -2,40 +2,65 @@ import logo from './logo.svg';
...
+    return `Error! ${error.message}`;
   }
 
-  return users.filter((user) => {
-    console.log("Query is: " + query);
+  //return <p>{  !loading && !error && data && JSON.stringify(data) }</p>;
```
Comment (zhangxian97) [inline basicapp/src/App.js:38]: same.

File: basicapp/src/App.js [hunk 4]
```diff
@@ -2,40 +2,65 @@ import logo from './logo.svg';
...
+  if (!loading && !error && data){
+    return (<Combobox
+              filter='contains'
+              dataKey='id'
+              textField='username'
+              value={query}
+              onChange={query => setQuery(query)}
+              //onSelect={alert('hi there ')}
+              //onSelect={() => {alert('hi there ');}}
+              onSelect={query => { alert('hi there ' + query.username);}}
```
Comment (zhangxian97) [inline basicapp/src/App.js:50]: same. remove comments.

Also don't use alert() in web development.

**Review by zhangxian97 — APPROVED** [review]
1 nit and some comments to remove.

**Pull request merged**